Documentation/using-corert/reflection-in-aot-mode.md

```diff
@@ -0,0 +1,65 @@
+# Reflection in AOT mode #
+
+When .NET code is compiled ahead of time, a typical problem the ahead of time compiler faces is deciding what code to compile and what data structures to generate.
+
+For static languages such as C or C++, the problem of deciding what to compile ahead of time is quite simple: one starts with compiling `Main()` and establishing what other methods and data structures `Main()` references. One then compiles those references, the references of the references and so on, until there's nothing left to compile. This concept is easy to understand and works great for languages like C or C++. Nice side effect of this approach is that the generated program is small. If the code doesn't call into e.g. the `printf` function, the `printf` function is not generated in the final executable.
+
+Problems with such approach start to show up on platforms that allow reflection. Reflection is a mechanism .NET provides that allows developers to inspect the structure of the program at runtime and access/invoke types and their members.
+
+As a motivating example, consider this program:
+
+```csharp
+class Program
+{
+    public static void Main()
+    {
+        Console.Write("Name of type: ");
+        string typeName = Console.ReadLine();
+        if (String.IsNullOrEmpty(typeName))
+            return;
+
+        Console.Write("Name of method: ");
+        string methodName = Console.ReadLine();
+
+        Type.GetType(typeName).GetMethod(methodName).Invoke(null, null);
+    }
+
+    public static void SayHello()
+    {
+        Console.WriteLine("Hello!");
+    }
+}
+
+```
+
+The above program lets the user invoke any parameterless public static method on any type. For the naive compilation algorithm above, this program would work great for input strings `Program` and `Main` because the algorithm included method `Main` in the final executable. The program wouldn't work so great for inputs `Program` and `SayHello`*, because method `SayHello` wasn't called from anywhere. For the naive algorithm, the only way to fix the program for inputs `Program` and `SayHello` is to add a call to `SayHello` in `Main`.
+
+> * The behavior for `Type.GetMethod` on type `Program` and method `SayHello` would be to return `null` if `SayHello` wasn't compiled. The reason for this is that `Type.GetMethod` is documented to return `null` if there's no method with a given name, and for the purposes of the program, `SayHello` doesn't exist. The compiler could remember there used to be such method and write the information in the executable, but that would raise additional questions about whether the uncompiled method should be included in a `Type.GetMethods` call.
```

Comment:
Nit: there's some weird formatting here as you have `>` and `*`, which quotes the paragraph and adds a bullet point (https://github.com/dotnet/corert/blob/aae75f266771b82bf61bffbca0c12793cb0a828a/Documentation/using-corert/reflection-in-aot-mode.md), I think you just want `> The behaviour for ...`

Reply:
Fixed.